app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,99 @@
+class FriendshipsController < ApplicationController
+  include LolConnections
+  before_action :search_summoner_for_request_action, only: [:edit]
+  before_action :require_user
+  before_action :require_same_user, only: [:edit, :decide]
+  
+ 
+	def new
+    @user = current_user
+    @friend = User.find_by_name(params[:summoner_to_add].downcase)
+    @friendship = Friendship.new(:user_id => @user.id, :friend_id => @friend.id, :initiator => true)
+    reverse_friendship = Friendship.new(:user_id => @friend.id, :friend_id =>  @user.id)
+    if (@friendship.save && reverse_friendship.save)
+      @friendship.send_friend_request_email
+      flash[:info] = "Summoner request sent."
+      redirect_to profile_path(@user.id)
+    else
+      flash[:danger] = @friendship.errors.messages[:base][0]
+      redirect_to :back
+    end
+  end
+
+  def edit
+    @friendship = @user.friendships_not_initiated_by_me.find_by(friend_id: @requester.id)
+    if @friendship.request_responded_at != nil
+      flash[:danger] = "You have either accepted or denied that request. If you'd like to add summoner, send a new LeagueSpec request"
+      redirect_to profile_path(@user.id)
+    end
+    
+  end
+
+  def index
+    @friends_list_code = params[:friends_list_code]
+    @user = current_user
+    if @friends_list_code == 'all'
+      @friend_count = @user.accepted_friendships.count
+      @friendships = @user.friendships.accepted.paginate(page: params[:page], per_page: 5)
+    elsif @friends_list_code == 'active'
+      @friendships = @user.get_online_friends.paginate(page: params[:page], per_page: 5)
+    else 
+      @pending_friendships_count = @user.pending_friendships.count
+      @friendships = @user.friendships.pending.paginate(page: params[:page], per_page: 5)
+    end      
```

Comment:
Constants are specified with all caps.  We can ensure they are a constant by adding the `.freeze` method at the end of it.  `FRIENDS_LIST_LENGTH = 5.freeze`.  Also, you definitely don't want to make it a global variable since you will only be using it inside this class.  Just remove the $ sign and you should still have access to it anywhere in the controller since you declared it outside of a function.